lib/logstash/inputs/kinesis.rb

```diff
@@ -50,6 +50,9 @@ class LogStash::Inputs::Kinesis < LogStash::Inputs::Base
   # to enable the cloudwatch integration in the Kinesis Client Library.
   config :metrics, :validate => [nil, "cloudwatch"], :default => nil
 
+  # Select AWS profile for input
+  config :profile, :validate => :string, :default => ""
```

Comment:
I recommend not setting a default, here. Then, below, instead of @profile.empty? you can do @profile.nil?